qa/tasks/cephfs/test_volumes.py

```diff
@@ -6603,6 +6483,267 @@ def test_clean_stale_subvolume_snapshot_metadata(self):
         self.mount_a.run_shell(['sudo', 'rm', '-f', tmp_meta_path], omit_sudo=False)
 
 
+class TestSubvolumeSnapshotGetpath(TestVolumesHelper):
+
+    def get_subvol_uuid(self, subvol_name, group_name=None):
+        '''
+        Return the UUID directory component obtained from the path of
+        subvolume.
+        '''
+        if group_name:
+            cmd = (f'fs subvolume getpath {self.volname} {subvol_name} '
+                   f'{group_name}')
+        else:
+            cmd = f'fs subvolume getpath {self.volname} {subvol_name}'
+
+        subvol_path = self.get_ceph_cmd_stdout(cmd).strip()
+
+        subvol_uuid = os.path.basename(subvol_path)
+        return subvol_uuid
+
+    def construct_snap_path(self, subvol_name, snap_name, uuid=None,
+                            group_name='_nogroup', v1=False):
+        if v1:
+            return os.path.join('/volumes', group_name, subvol_name, uuid,
```

Comment:
if `uuid=None` won't this fail every time?

Reply:
I've split this method into 3 (each for legacy, v1 and v2) so that there are no such corner cases.